### PR TITLE
Qualcomm AI Engine Direct - LLM multi-batch quantization and evaluation

### PR DESCRIPTION
**Pull request by DannyYuyang-quic**
### Summary
- Support multi-batch calibration
- Support multi-batch evaluation
- Add CLI flag to specify batch size during calibration
- Fix the shape of attention mask (Optimize runtime graph)
  
### Details about fix the shape of attention mask (Optimize runtime graph)
The attention mask without the head dimension, the model had to broadcast the shape from [1, 1, S] to [1, 1, 1, S]. This introduced an extra `view_copy` node in the runtime graph, which is unnecessary.
image
With head dim, no broadcast needed, redundant `view_copy` node removed
image

### Test plan
- ExampleLLMScript
- TestExampleMultimodalityScript

### CODE REVIEW

**Comment by DannyYuyang-quic** [comment]
@psiddh Hi, this PR mainly adds support for multi-batch quantization and evaluation.
Please take a look. Thanks!

cc: @shewu-quic, @haowhsu-quic

**Comment by DannyYuyang-quic** [comment]
@psiddh Hi, just wanted to give this PR a gentle nudge when you have a chance. 
I'd appreciate your review. Thanks!!

File: examples/qualcomm/oss_scripts/llama/evaluator/lm_eval_adapter.py [hunk 1]
```diff
@@ -43,15 +45,23 @@ def __init__(
...
+
+    @property
+    def batch_size(self):
+        return self._batch_size
 
     def _model_call(self, inps):
-        logits = self._runner.predict_step(
+        actual_bsz = inps.shape[0]
+        inps = F.pad(inps, (0, 0, 0, self._batch_size - actual_bsz))
+        logits = self._runner.prediction_step(
```
Comment (psiddh) [inline examples/qualcomm/oss_scripts/llama/evaluator/lm_eval_adapter.py]: Should this be predict_step ?  DecoderInference defines predict_step (decoder.py:136) and every other call site uses it, so prediction_step, not sure , can you pls clarify ?

File: examples/qualcomm/oss_scripts/llama/runner/runner.cpp [hunk 2]
```diff
@@ -244,10 +244,10 @@ Error Runner::load() {
   int32_t token_generator_ar_len = 0;
   int32_t max_cache_len = 0;
   int32_t max_ar_len = 0;
-  // atten mask: [1, AR-N, CL]
+  // atten mask: [1, 1, AR-N, CL]
```
Comment (psiddh) [inline examples/qualcomm/oss_scripts/llama/runner/runner.cpp:247]: Since the mask goes from 3-D to 4-D and the runner now reads sizes()[2]/[3], a .pte exported before this change won't match the new runner (and vice versa), right?  am I missing something ?

File: examples/qualcomm/oss_scripts/llama/evaluator/lm_eval_adapter.py (hunk 1 above)
Comment (DannyYuyang-quic) [reply examples/qualcomm/oss_scripts/llama/evaluator/lm_eval_adapter.py]: Oh yes, it should be `predict_step` that's a typo. Thanks for catching this.

File: examples/qualcomm/oss_scripts/llama/runner/runner.cpp (hunk 2 above)
Comment (DannyYuyang-quic) [reply examples/qualcomm/oss_scripts/llama/runner/runner.cpp:247]: Yes, the new runner couldn't run the generated .pte before this change. That's also why I updated stories260.pte for the Backward Compatibility CI tests.

File: examples/qualcomm/oss_scripts/llama/evaluator/lm_eval_adapter.py (hunk 1 above)
Comment (DannyYuyang-quic) [reply examples/qualcomm/oss_scripts/llama/evaluator/lm_eval_adapter.py]: Fixed!

**Review by psiddh — APPROVED** [review]
